### PR TITLE
Update to mamba 2.0.5

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Why is removing the boa compatibility check ok? 

Has boa been removed completely from our tooling that we use on feedstocks?

has boa been deprecated upstream?

im really not sure. Just asking.

**Comment by SylvainCorlay** [comment]
Since 2.0.5 is coming out now, could we directly upgrade to 2.0.5?

File: Miniforge3/construct.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {% set name = os.environ.get("MINIFORGE_NAME", "Miniforge3") %}
-{% set version = os.environ.get("MINIFORGE_VERSION", "24.11.0-0") %}
-{% set conda_libmamba_solver_version = "24.9.0"%}
+{% set version = os.environ.get("MINIFORGE_VERSION", "24.11.0-1") %}
+{% set conda_libmamba_solver_version = "24.11.1"%}
 # when mamba_version is updated here, also update MICROMAMBA_VERSION in scripts/build.sh
-{% set mamba_version = "1.5.11"%}
+{% set mamba_version = "2.0.4"%}
```
Comment (jjerphan) [inline Miniforge3/construct.yaml:5]:
```suggestion
{% set mamba_version = "2.0.5"%}
```

**Review by jjerphan — COMMENTED** [review]
When 2.0.5 becomes available on CDNs.

**Comment by hmaarrfk** [comment]
Thanks. Release started.

**Pull request merged**